pull.go

```diff
@@ -152,60 +152,60 @@ func localImageNameForReference(ctx context.Context, store storage.Store, srcRef
 
 // Pull copies the contents of the image from somewhere else to local storage.
 func Pull(ctx context.Context, imageName string, options PullOptions) error {
-	spec := imageName
 	systemContext := getSystemContext(options.SystemContext, options.SignaturePolicyPath)
-	srcRef, err := alltransports.ParseImageName(spec)
```

Comment:
If I am reading `return []string{split[1]}, false, nil` in `ResolveName` correctly, this is now broken — it only works if `options.Transport` matches the transport in the input.

_That_ problem, in turn, is masked by e.g. `buildah from` parsing the input string and setting `options.Transport` to make this all work — but non-CLI callers could be tripped by this, and it’s a very fragile way to structure the code anyway.